### PR TITLE
disable nullability strictness

### PR DESCRIPTION
**Pull request by sdelamo**
see: https://github.com/micronaut-projects/micronaut-aws/pull/1771

### CODE REVIEW

**Review by dstepanov — CHANGES_REQUESTED** [review]

File: serde-api/src/main/java/io/micronaut/serde/Deserializer.java [hunk 1]
```diff
@@ -37,6 +37,8 @@
 @Indexed(Deserializer.class)
 public interface Deserializer<T> {
 
+    boolean STRICT_NULLABLE = false;
```
[the remark this answers is not in the document]
Comment (sdelamo) [reply serde-api/src/main/java/io/micronaut/serde/Deserializer.java]: Good idea. [I moved it to DeserializationConfiguration](https://github.com/micronaut-projects/micronaut-serialization/pull/489/commits/56a0653829fad0d071639aa6665c461d70db7cb8)

**Review by dstepanov — APPROVED** [review]

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java [hunk 2]
```diff
@@ -313,7 +314,7 @@ public Object deserialize(Decoder decoder, DecoderContext decoderContext, Argume
                 }
             } else {
                 try {
-                    obj = db.introspection.instantiate();
+                    obj = db.introspection.instantiate(strictNullable, new Object[] {});
```
Comment (graemerocher) [inline serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java:317]: use `ArrayUtils.EMPTY_OBJECT_ARRAY`

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java [hunk 3]
```diff
@@ -63,7 +62,9 @@ public Object deserialize(Decoder decoder, DecoderContext decoderContext, Argume
                 final PropertiesBag<Object>.Consumer creatorParams = db.creatorParams.newConsumer();
                 final DeserBean.DerProperty<Object, Object> creator = creatorParams.getNotConsumed().iterator().next();
                 final Object val = deserializeValue(decoderContext, decoder, creator, creator.argument, null);
-                return db.introspection.instantiate(val);
+                Object[] args = new Object[1];
```
Comment (graemerocher) [inline serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java:65]: initialise on the same line `new Object[] { val }`

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java [hunk 4]
```diff
@@ -664,9 +665,10 @@ private void applyDefaultValuesOrFail(
                         }
                     }
                 }
-                object = unwrapped.introspection.instantiate(params);
+
+                object = unwrapped.introspection.instantiate(strictNullable, params);
             } else {
-                object = unwrapped.introspection.instantiate();
+                object = unwrapped.introspection.instantiate(strictNullable, new Object[] {});
```
Comment (graemerocher) [inline serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java:671]: use `ArrayUtils.EMPTY_OBJECT_ARRAY`

File: serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java [hunk 5]
```diff
@@ -145,12 +142,16 @@ public EnumCreatorDeserializer(
...
+            Object[] args = new Object[1];
+            args[0] = v;
+            return (E) deserializableIntrospection.instantiate(!allowNull, args);
         } catch (IllegalArgumentException e) {
             if (v instanceof String) {
                 String string = (String) v;
                 try {
-                    return (E) deserializableIntrospection.instantiate(string.toUpperCase(Locale.ENGLISH));
+                    Object[] args = new Object[1];
+                    args[0] = string.toUpperCase(Locale.ENGLISH);
```
Comment (graemerocher) [inline serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java:153]: initialise on the same line `new Object[] { val }`

File: serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java [hunk 6]
```diff
@@ -145,12 +142,16 @@ public EnumCreatorDeserializer(
     @NonNull
     private E transform(Object v) {
         try {
-            return (E) deserializableIntrospection.instantiate(v);
+            Object[] args = new Object[1];
```
Comment (graemerocher) [inline serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java:145]: initialise on the same line `new Object[] { val }`

File: test-suite/build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+plugins {
```
Comment (graemerocher) [inline test-suite/build.gradle.kts:1]: do we need a whole new test suite for this? Could we not have added to the already existing tests

**Review by graemerocher — COMMENTED** [review]
@sdelamo I didn't get a chance to review this before you merged, can you address my feedback please

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java (hunk 3 above)
Comment (sdelamo) [reply serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java:65]: done via https://github.com/micronaut-projects/micronaut-serialization/pull/492

File: serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java (hunk 5 above)
Comment (sdelamo) [reply serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java:153]: Done via https://github.com/micronaut-projects/micronaut-serialization/pull/492

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java (hunk 4 above)
Comment (sdelamo) [reply serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java:671]: Done via https://github.com/micronaut-projects/micronaut-serialization/pull/492

File: test-suite/build.gradle.kts (hunk 7 above)
Comment (sdelamo) [reply test-suite/build.gradle.kts:1]: done via https://github.com/micronaut-projects/micronaut-serialization/pull/492

**Pull request merged**